projects/exui-common-lib/src/lib/components/search-location/search-location.component.ts

```diff
@@ -147,4 +148,16 @@ export class SearchLocationComponent implements OnInit {
       location => !selectedLocations.map(selectedLocation => selectedLocation.epimms_id).includes(location.epimms_id) && location[this.propertyNameFilter]
     );
   }
+
+  public RemoveInvalidString(formInputValue: any) {
+    const element = formInputValue.target as HTMLInputElement;
+    const newInputValue = element.value;
```

Comment:
Why is this.binaryValue called "binaryValue". It looks like it is the previous value

Keeping track of the previous value  is dependent on keyboard input being the only mechanism for changing the value of the field, which I don't think we can guarantee. It also probably won't work for fields that have an initial value.
The keyup event will tell which key has been input via the event parameter and you should processing this, not the value of the field. 

Read the docs https://angular.io/guide/user-input
you want something like this:
onKeyRemoveInvalid(event: KeyboardEvent) {
   if (!isValidNameCharacter(event)
      event.preventDefault()
  }
  function isValidNameCharacter(event: KeyboardEvent): boolean {
    let pressed = undefined
    if (event.key !== undefined) {
        pressed = event.key; // might be "Enter" or "Shift"
        if (pressed.length > 1) return false; //filter out control characters
    } else if (event.keyCode !== undefined) {
        pressed = String.fromCharCode(event.keyCode); //
    }
    return pressed && pressed.match(/[A-Za-z0-9\s\'\-]/));   
  }
}
}
You need to include ' as valid char e.g. O'Malley

Reply:
@andywilkinshmcts , I have made some few changes to my code.

However, I believe that keeping track of the keyboard input is, in this instance, the right thing to do. The part of the application upon which this code depends (the location filter) waits for user input and then makes an http call when the input is greater than 3 characters. We do not require an Enter, Shift events for this functionality.

I am happy to discuss/clarify this further on the phone when you're free.